modules/sequence/src/main/scala/lucuma/odb/sequence/util/SequenceIds.scala

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.sequence.util
+
+import lucuma.core.enums.SequenceType
+import lucuma.core.model.Observation
+import lucuma.core.model.sequence.Atom
+import lucuma.core.model.sequence.Step
+import lucuma.core.util.Uid
+import lucuma.odb.sequence.data.GeneratorParams
+
+import java.io.ByteArrayOutputStream
+import java.io.ObjectOutputStream
+import java.util.UUID
+import scala.util.Using
+
+
+object SequenceIds {
+
+  def namespace(
+    observationId: Observation.Id,
+    params:        GeneratorParams
+  ): UUID =
```

Comment:
A potential point of discussion here.  The sequence IDs are generated using "type 3" UUIDs (i.e., a hash of input bytes).  I'd originally thought to generate the sequence without IDs and pass those bytes into `UUID.nameUUIDFromBytes`.  It occurs to me though that the sequence generation is deterministic from the generator params and observation id so in theory that is (almost) enough.  We'd need to take into account changes to the generator code itself by incorporating a GIT hash I think and in the ITC using the version information it returns with results.

An added benefit is that we can LRU cache the generated sequence by the "namespace" UUID.

Reply:
:+1: based on Slack discussion this makes sense. You can't get to git info in the Heroku build but you can get the commit via (dyno-metadata)[https://devcenter.heroku.com/articles/dyno-metadata], which is a blunt object but it should work.

Reply:
Added a `CommitHash` which is taken from the metadata as discussed.